src/vs/base/browser/ui/button/button.ts

```diff
@@ -307,168 +317,41 @@ export class ButtonWithDropdown extends Disposable implements IButton {
-				this.setHoverBackground();
-			}
-		}));
-
-		this._register(addDisposableListener(this._element, EventType.MOUSE_OUT, e => {
-			this.applyStyles(); // restore standard styles
-		}));
-
-		// Also set hover background when button is focused for feedback
-		this.focusTracker = this._register(trackFocus(this._element));
-		this._register(this.focusTracker.onDidFocus(() => this.setHoverBackground()));
-		this._register(this.focusTracker.onDidBlur(() => this.applyStyles())); // restore standard styles
-
-		this.applyStyles();
-	}
-
-	private setHoverBackground(): void {
-		let hoverBackground;
-		if (this.options.secondary) {
-			hoverBackground = this.buttonSecondaryHoverBackground ? this.buttonSecondaryHoverBackground.toString() : null;
-		} else {
-			hoverBackground = this.buttonHoverBackground ? this.buttonHoverBackground.toString() : null;
-		}
-		if (hoverBackground) {
-			this._element.style.backgroundColor = hoverBackground;
-		}
-	}
-
-	style(styles: IButtonStyles): void {
-		this.buttonForeground = styles.buttonForeground;
-		this.buttonBackground = styles.buttonBackground;
-		this.buttonHoverBackground = styles.buttonHoverBackground;
-		this.buttonSecondaryForeground = styles.buttonSecondaryForeground;
-		this.buttonSecondaryBackground = styles.buttonSecondaryBackground;
-		this.buttonSecondaryHoverBackground = styles.buttonSecondaryHoverBackground;
-		this.buttonBorder = styles.buttonBorder;
-
-		this.applyStyles();
-	}
-
-	private applyStyles(): void {
-		if (this._element) {
-			let background, foreground;
-			if (this.options.secondary) {
-				foreground = this.buttonSecondaryForeground ? this.buttonSecondaryForeground.toString() : '';
-				background = this.buttonSecondaryBackground ? this.buttonSecondaryBackground.toString() : '';
-			} else {
-				foreground = this.buttonForeground ? this.buttonForeground.toString() : '';
-				background = this.buttonBackground ? this.buttonBackground.toString() : '';
-			}
-
-			const border = this.buttonBorder ? this.buttonBorder.toString() : '';
-
-			this._element.style.color = foreground;
-			this._element.style.backgroundColor = background;
-
-			this._element.style.borderWidth = border ? '1px' : '';
-			this._element.style.borderStyle = border ? 'solid' : '';
-			this._element.style.borderColor = border;
-		}
-	}
-
-	get element(): HTMLElement {
-		return this._element;
+		this.element.appendChild(this._descriptionElement);
 	}
 
 	set label(value: string) {
-		this._element.classList.add('monaco-text-button');
+		this.element.classList.add('monaco-text-button');
 		if (this.options.supportIcons) {
 			reset(this._labelElement, ...renderLabelWithIcons(value));
 		} else {
 			this._labelElement.textContent = value;
 		}
 		if (typeof this.options.title === 'string') {
-			this._element.title = this.options.title;
+			this.element.title = this.options.title;
 		} else if (this.options.title) {
-			this._element.title = value;
+			this.element.title = value;
 		}
 	}
```

Comment:
Is there a reason this setter still has to be duplicated between the two concrete implementations of GenericButton subclasses? This PR does remove a lot of duplicate code, but _this_ function gets turned into an exact duplicate of the setter in the `Button` class, which is kind of a bummer. :disappointed: